atsc/src/data.rs

```diff
@@ -70,21 +72,34 @@ impl CompressedStream {
         }
         compressor_frame.close();
         self.data_frames.push(compressor_frame);
+        self.header.add_frame();
     }
 
     /// Transforms the whole CompressedStream into bytes to be written to a file
     pub fn to_bytes(self) -> Vec<u8> {
-        // Will this chain encode??
+        let mut out = Vec::new();
         let config = BinConfig::get();
-        bincode::encode_to_vec(self, config).unwrap()
+        out.extend_from_slice(&self.header.to_bytes());
+        out.extend_from_slice(&bincode::encode_to_vec(self.data_frames, config).unwrap());
```

Comment:
this line will write the body of the file (99% of the file) into one buffer and then copy the result of that into another buffer (`out`) and most likely trigger an allocation in the process.

To avoid this, I think we can instead write the line as `bincode::encode_into_std_write(self.data_frames, &mut out, config).unwrap();`.
I havent tried compiling this though.